tests/training/test_train_sac_sb3_config.py

```
@@ -0,0 +1,431 @@
+        assert "pedestrians_positions" in vec_env.observation_space.spaces
+        assert isinstance(vec_env.action_space, gym_spaces.Box)
+        assert vec_env.action_space.shape == (2,)
+    finally:
+        vec_env.close()
+
+
+def test_build_env_switches_across_loaded_scenarios(monkeypatch: pytest.MonkeyPatch) -> None:
+    """SAC training env should sample across the loaded scenario list per reset."""
+
+    class _DummyEnv(gymnasium.Env):
+        metadata: dict[str, object] = {}
```

Ruff (0.15.9)

[warning] Mutable default value for class attribute

(RUF012)